src/Raven.Server/ServerWide/SubscriptionsClusterStorage.cs

```diff
@@ -25,6 +27,18 @@ public SubscriptionState ReadSubscriptionStateByName(ClusterOperationContext con
             throw new SubscriptionDoesNotExistException($"Subscription with name '{name}' was not found in server store");
 
         var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+        
+        if (subscriptionState.ArchivedDataProcessingBehavior.HasValue)
+            return subscriptionState;
+        
+        // persisted state from v5.x
+        if(Enum.TryParse(_cluster.ReadDatabase(context, databaseName).Settings[RavenConfiguration.GetKey(x => x.Subscriptions.ArchivedDataProcessingBehavior)], false,
+               out ArchivedDataProcessingBehavior archivedDataProcessingBehavior) == false)
+        {
+            throw new InvalidOperationException(
```

Comment:
I'm not sure if I follow - if we fail to read it from the database record then we throw? That will be always the case for subscriptions from version 5.x, won't it?

Reply:
not exactly, explained directly